aiidalab_widgets_base/structures.py

```diff
@@ -4,126 +4,113 @@
+    Attributes:
+        structure(Atoms): trait that contains the selected structure. 'None' if no structure is selected.
+        structure_node(StructureData, CifData): trait that contains AiiDA structure object
+        node_class(str): trait that contains structure_node type (as string).
+    '''
+
+    structure = Union([Instance(Atoms), Instance(Data)], allow_none=True)
+    structure_node = Instance(Data, allow_none=True, read_only=True)
+    node_class = Unicode()
 
-    has_structure = Bool(False)
-    frozen = Bool(False)
-    DATA_FORMATS = ('StructureData', 'CifData')
+    DATA_FORMATS = {'CifData': 'cif', 'StructureData': 'structure'}
 
     def __init__(self, importers, storable=True, node_class=None, **kwargs):
         """
-        :param storable: Whether to provide Store button (together with Store format)
-        :type storable: bool
-        :param node_class: AiiDA node class for storing the structure.
-            Possible values: 'StructureData', 'CifData' or None (let the user decide).
-            Note: If your workflows require a specific node class, better fix it here.
-        :param examples: list of tuples each containing a name and a path to an example structure
-        :type examples: list
-        :param importers: list of tuples each containing a name and an object for data importing. Each object
-        should containt an empty `on_structure_selection()` method that has two parameters: structure_ase, name
-        :type examples: list"""
+        Arguments:
+            importers(list): list of tuples each containing the displayed name of importer and the
+                importer object. Each object should containt 'structure' trait pointing to the imported
+                structure. The trait will be linked to 'structure' trait of this class.
+
+            storable(bool): Whether to provide Store button (together with Store format)
+
+            node_class(str): AiiDA node class for storing the structure.
+                Possible values: 'StructureData', 'CifData' or None (let the user decide).
+                Note: If your workflows require a specific node class, better fix it here.
+        """
 
         from .viewers import StructureDataViewer
-        if not importers:  # we make sure the list is not empty
+
+        # Make sure the list is not empty
+        if not importers:
             raise ValueError("The parameter importers should contain a list (or tuple) of tuples "
                              "(\"importer name\", importer), got a falsy object.")
 
-        self.structure_ase = None
-        self._structure_node = None
+        # Store button.
+        self.btn_store = ipw.Button(description='Store in AiiDA', disabled=True)
+        self.btn_store.on_click(self._on_click_store)
+
+        # Setting traits' initial values
+        self.structure = None
+        self._inserted_structure = None
+        self.set_trait('structure_node', None)
```

Comment:
I think it is best practice to use [`@default` functions](https://traitlets.readthedocs.io/en/stable/using_traitlets.html#default-values-and-checking-type-and-value) for that.

Reply:
thanks!